pkg/cloudscale/objectstorage.go

```diff
@@ -79,37 +108,45 @@ func (o *ObjectStorage) GetMetrics(ctx context.Context, billingDate time.Time) (
 
 	buckets, err := fetchBuckets(ctx, o.k8sClient)
 	if err != nil {
+		o.providerMetrics["providerFailed"].Inc()
 		return nil, err
 	}
 
-	allRecords := make([]odoo.OdooMeteredBillingRecord, 0)
-	for _, bucketMetricsData := range bucketMetrics.Data {
-		name := bucketMetricsData.Subject.BucketName
-		logger = logger.WithValues("bucket", name)
-		bd, ok := buckets[name]
-		if !ok {
-			logger.Info("unable to sync bucket, ObjectBucket not found")
-			continue
+	for bucket := range bucketMap {
```

Comment:
Why don't you get here the key and the value? You will not need to search for the value subsequently.

Reply:
wdym?

Reply:
is it really that important to You? ;d

Reply:
well apart from the fact that it's less performant which in this case is really not much, my problem is that it was confusing to read the logic, I was asking myself why it's done that way and was trying harder to understand maybe I am missing something. We need to make the code simple and intuitive otherwise we get more bugs.

Reply:
I made a small refactor, so You can smile at work :P